src/ipfs-pubsub.js

```diff
@@ -70,8 +70,14 @@ class IPFSPubsub {
   }
 
   publish(topic, message, options = {}) {
-    if(this._subscriptions[topic] && this._ipfs.pubsub) {
-      this._ipfs.pubsub.publish(topic, Buffer.from(JSON.stringify(message)), options)
+    if (this._subscriptions[topic] && this._ipfs.pubsub) {
+      var payload;
+      if(typeof message === "object") {
+        payload = JSON.stringify(message);
+      } else {
+        payload = message;
```

Comment:
What is the type here,then? `string`, `Buffer`? Maybe there is a way to make it explicit for the reader, maybe even just a comment?

Reply:
Ohh Yes you are right../..

Reply:
Buffer would be the only case where it is treated as an object. String has its own type.